ebos/eclproblem.hh

```diff
@@ -3147,6 +3153,33 @@ private:
         return dtNext;
     }
 
+    void computeAndSetEqWeights_() {
+        std::vector<Scalar> sumInvB(numPhases, 0.0);
+        size_t numDof = this->model().numGridDof();
+        for (size_t dofIdx = 0; dofIdx < numDof; ++dofIdx) {
+            const auto& dofFluidState = initialFluidStates_[dofIdx];
+            for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
+                if (!FluidSystem::phaseIsActive(phaseIdx))
+                    continue;
+
+                sumInvB[phaseIdx] += dofFluidState.invB(phaseIdx);
+            }
+        }
+        const auto& comm = this->simulator().vanguard().grid().comm();
+        comm.sum(sumInvB.data(),sumInvB.size());
+        Scalar numTotalDof = comm.sum(numDof);
```

Comment:
I assume that these are all dofs regardless whether they are at ghost cells or not. Then both sumInvB and numTotalDof will be too high in parallel as dofs that are in the ghost area on a process will be counted twice.

The easiest fix would be to iterate over the interior of the grid. Not sure whether there is a way to get the corresponding dof indices then.

Reply:
This should be fixed now